library/core/src/main/java/com/google/android/exoplayer2/upstream/cache/CacheDataSink.java

```diff
@@ -70,9 +71,10 @@ public CacheDataSinkException(IOException cause) {
    * @param maxCacheFileSize The maximum size of a cache file, in bytes. If the sink is opened for
    *    a {@link DataSpec} whose size exceeds this value, then the data will be fragmented into
    *    multiple cache files.
+   * @param skipFDSync Skip file descriptor sync when closing current output stream.
```

Comment:
Could we add keep a 2 argument constructor, to avoid having to update all the calling sites, which is most of the rest of this change?

Also, the more I think about this the more I think it'll be fine to eventually default to not doing the sync. What I'd propose is that the two arg constructor defaults to doing to sync for now, we'll experiment with turning the sync off for some internal apps we have, and flip it at some point in the future if we don't see any increase in corruption issues.

Given the plan to eventually default to not doing the sync, it's IMO preferable if the argument is flipped from what it is now (i.e. called `syncFileDescriptors`), so the eventual default will be `false`.

Reply:
Thanks! I will update the change accordingly.